src/States/State.php

```diff
@@ -10,10 +10,21 @@
 
 abstract class State extends BaseState implements Arrayable
 {
-    protected static ?StateConfig $config = null;
+    protected static ?array $config = null;
 
     protected static ?string $color = null;
 
+    public static function registerStateConfig(StateConfig $config, ?string $baseStateClass = null): void
+    {
+        if (! is_a($baseStateClass ?? $config->baseStateClass, static::class, true)) {
+            throw new \InvalidArgumentException(
+                "The state class `{$config->baseStateClass}` must be a subclass of `" . static::class . '`'
```

Comment:
exception does not necessarily contain the given state class. $baseStateClass ?? $config->baseStateClass
Also no need to extra concat static::class, can just pass static::class the same way as baseStateClass into the string.